inference/sources/fly/main_test.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	. "github.com/vlab-research/vlab/inference/inference-data"
+	. "github.com/vlab-research/vlab/inference/test-helpers"
+)
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func resData(fi string) string {
+	b, e := ioutil.ReadFile(fmt.Sprintf("test/%s", fi))
+	check(e)
+
+	return string(b)
+}
+
+func normalizeSpace(s string) string {
+	re := regexp.MustCompile(`\s+`)
+	r := re.Split(s, -1)
+	return strings.Join(r, " ")
+}
+
+func dataAssertions(t *testing.T, e []*InferenceDataEvent) {
+	assert.Equal(t, 2, len(e))
+
+	for i := 0; i < len(e); i++ {
+		assert.Equal(t, 1, e[0].Idx)
+		assert.Equal(t, 2, e[1].Idx)
+	}
+}
+func TestGetResponses_PaginatesWhenPageIsFull(t *testing.T) {
+	res1 := resData("fly_example.json")
+	res2 := resData("fly_example2.json")
+
+	s := string(res1)
+	data := GetResponsesResponse{}
+	json.Unmarshal([]byte(s), &data)
+
+	s2 := string(res2)
+	data2 := GetResponsesResponse{}
+	json.Unmarshal([]byte(s2), &data2)
+
+	count := 1
+	// Supongamos que 2 es el limite de items por page
+	full_page_with_items := 2
+	ts, _ := TestServer(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/all", r.URL.Path)
+		if count == 1 {
+			assert.Equal(t, "after=be5ae9dd-0189-478e-8a3d-4d8ead8240a4&page_size=1", r.URL.RawQuery)
+			page_size, err := strconv.Atoi(r.URL.Query().Get("page_size"))
+
+			if err != nil {
+				log.Fatal(err)
+			}
+			assert.Equal(t, count, page_size)
+
+			// PAGE #1
+			assert.Equal(t, count, data.PageCount)
```

Comment:
Not sure why you're asserting this...

Reply:
I was simulating that the page was filled with a certain amount of items and the counter simulates which page is currently on